timemachine/cpp/src/potential.cu

```diff
@@ -23,66 +25,63 @@ void Potential::execute_host(
 
     const int &D = Potential::D;
 
-    double *d_x;
-    double *d_p;
-    double *d_box;
-
-    gpuErrchk(cudaMalloc(&d_x, N * D * sizeof(double)));
-    gpuErrchk(cudaMemcpy(d_x, h_x, N * D * sizeof(double), cudaMemcpyHostToDevice));
-
-    gpuErrchk(cudaMalloc(&d_p, P * sizeof(double)));
-    gpuErrchk(cudaMemcpy(d_p, h_p, P * sizeof(double), cudaMemcpyHostToDevice));
+    DeviceBuffer<double> d_x(N * D);
+    DeviceBuffer<double> d_p(P);
+    DeviceBuffer<double> d_box(D * D);
 
-    gpuErrchk(cudaMalloc(&d_box, D * D * sizeof(double)));
-    gpuErrchk(cudaMemcpy(d_box, h_box, D * D * sizeof(double), cudaMemcpyHostToDevice));
+    gpuErrchk(cudaMemcpy(d_x.data, h_x, d_x.size, cudaMemcpyHostToDevice));
+    gpuErrchk(cudaMemcpy(d_p.data, h_p, d_p.size, cudaMemcpyHostToDevice));
+    gpuErrchk(cudaMemcpy(d_box.data, h_box, d_box.size, cudaMemcpyHostToDevice));
 
-    unsigned long long *d_du_dx = nullptr;
-    unsigned long long *d_du_dp = nullptr;
-    unsigned long long *d_du_dl = nullptr;
-    unsigned long long *d_u = nullptr;
+    std::unique_ptr<DeviceBuffer<unsigned long long>> d_du_dx;
```

Comment:
`unique_ptr` will automatically call the referenced `DeviceBuffer`'s destructor when the pointer goes out of scope.

Reply:
nice! that works too